src/ui/edit-page-link.tsx

```diff
@@ -7,10 +7,10 @@ import { useI18n } from "~/i18n/i18n-context";
 
 export const EditPageLink: Component = () => {
 	const i18n = useI18n();
-	const { pathname } = useLocation();
```

Comment:
Correct me if im wrong, but isn't `location.pathname` reactive? I'm not sure if it retains reactive properties when destructured, though 

That means we shouldn't need to memoize it.

Reply:
It's not reactive. Notice how pathname is used as pathname, not pathname().

Reply:
![Uploading Screenshot 2024-03-27 at 00.47.43.png…]()

Reply:
But keeping it as `location.pathname` vs destructuring it (ie. `{ pathname }`) should retain the reactive properties?

Reply:
`location` never had reactive properties. `location` is a normal object. This is i.e. the value of `location` on the `/` page:

`{"pathname":"/","search":"","hash":"","state":{"_depth":0},"key":"","query":{}}``

Reply:
This feels like it contradicts the docs on it - 
https://github.com/solidjs/solid-router?tab=readme-ov-file#uselocation

Reply:
Yeah, it's not explained that well. It can't be used as a reactive variable _inside the template_ unless a memo is introduced.

Reply:
Oh boy - I guess that means we need to clarify it more lol.

Reply:
Yes, we somehow have to explain how a `reactive function`, `reactive object` and `primitive` behave differently in the reactive graph. This is how I see their differences:

- `reactive function` (signal/memo) - It can be reactive in a template, and it can be used to produce new `reactive functions`, `reactive objects` and `primitives`.
- `reactive object` - If an object is derived from a `reactive function`, then it can be considered a `reactive object`. A `reactive object` isn't reactive when inserted into the template, but by wrapping it in a memo it can extend the reactive graph to a new `reactive function` that is.
- `primitive` (string/bool/number) - Can't be, or be made, reactive.